src/audio/codec_adapter/codec/cadence.c

```diff
@@ -278,6 +278,14 @@ int cadence_codec_prepare(struct comp_dev *dev)
 		goto err;
 	}
 
+	API_CALL(cd, XA_API_CMD_INIT, XA_CMD_TYPE_INIT_PROCESS, NULL, ret);
+	if (ret != LIB_NO_ERROR) {
+		comp_err(dev, "cadence_codec_prepare() error %x: failed to initialize codec",
+			 ret);
+		codec_free_memory(dev, cd->mem_tabs);
```

Comment:
I feel like you're calling codec_free_memory quite a bit, you might want to add a second err2: label?

Reply:
Yes I agree

Reply:
@mrajwa Looks like this comment is still not addressed. Sorry for us being picky, this is how a reviewer should be :).